File: app/src/main/java/fr/nihilus/mymusic/media/MediaItems.java
package fr.nihilus.mymusic.media;

import android.support.v4.media.MediaBrowserCompat;

/**
 * A helper class grouping {@link MediaBrowserCompat.MediaItem}-related constants.
 */
public final class MediaItems {
    /**
     * A non human-readable key used for sorting.
     * <p>Type: String</p>
     */
    public static final String EXTRA_TITLE_KEY = "title_key";
    /**
     * The listening time represented by this item, may it be a track, an album or a playlist.
     * <p>Type: long</p>
     */
    public static final String EXTRA_DURATION = "duration";
}

File: core/src/main/java/fr/nihilus/music/core/media/MediaItems.kt
/*
 * Copyright 2019 Thibault Seisel
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package fr.nihilus.music.core.media

/**
 * A helper class grouping media-related constants.
 */
object MediaItems {

    /**
     * The listening time represented by this item, may it be a track, an album or a playlist.
     *
     * Type: long
     */
    const val EXTRA_DURATION = "fr.nihilus.music.extra.DURATION"

    /**
     * The number of tracks that this browsable media item contains.
     *
     * Type: integer
     */
    const val EXTRA_NUMBER_OF_TRACKS = "fr.nihilus.music.extra.NUMBER_OF_TRACKS"

    /**
     * The number of this track in its album.
     *
     * Type: integer
     */
    const val EXTRA_TRACK_NUMBER = "fr.nihilus.music.extra.TRACK_NUMBER"

    /**
     * The number of the disc this track appears.
     *
     * Type: integer
     */
    const val EXTRA_DISC_NUMBER = "fr.nihilus.music.extra.DISC_NUMBER"
}
